include/seastar/coroutine/exception.hh

```diff
@@ -54,4 +84,23 @@ exception make_exception(T&& t) {
     return exception(std::make_exception_ptr(std::forward<T>(t)));
 }
 
-}
\ No newline at end of file
+/// Helper for creating a coroutine::exception instance. For clarity, it is
+/// recommended to be used along with co_await.
+///
+/// Example usage:
+///
+/// ```
+/// co_await coroutine::return_exception(std::runtime_error("something failed miserably"));
+/// ```
+template<typename T>
+exception return_exception(T&& t) {
```

Comment:
add nodiscard here to prevent `coroutine::return_exception(...);` from succeeding.

The documentation is accurate but not helpful. The goal is to return an exception from a coroutine, the user has no goal to create a coroutine::exception. co_await is required, not recommended. Write the documentation from the user point of view, don't describe the internals.

Reply:
Added nodiscard to `return_exception`, improved the docs.

Should I add nodiscard to `make_exception`, too? I was hesitant to do it because technically it would be a backwards-incompatible change, but only for the code which uses `make_exception` incorrectly. Alternatively, I can just mark the `exception` nodiscard, then nodiscard will be unnecessary for `return_exception` and `make_exception`.

Reply:
I think it's (marginally) better to annotate {make,return}_exception. The closer the error location to the caller, the easier it is to understand what's going on.

We don't need to keep compatibility with broken code.

Reply:
OK, added `[[nodiscard]]` to `make_exception`, too.